common/app/routes/challenges/redux/selectors.js

```diff
@@ -28,9 +28,11 @@ export const challengeSelector = createSelector(
         submitTypes[challenge && challenge.type] ||
         'tests',
       showPreview: challengeType === html,
+      /* eslint-disable no-nested-ternary */
       mode: challenge && challengeType === html ?
-        'text/html' :
+        'text/html' : challengeType === 8 ? 'jsx' :
```

Comment:
`challengeType` is deprecated. We should be using the modern view for all new challenge types where applicable. This are challenges with `type: 'modern'`.